Library/Homebrew/os/mac/mach.rb

```diff
@@ -80,7 +80,7 @@ def mach_data
       when 0x7f454c46 # ELF
         mach_data << { :arch => :x86_64, :type => :executable }
       else
-        raise "Not a Mach-O binary."
+        return []
```

Comment:
Doesn't the `return` circumvent the caching that happens in this method (it starts with `@mach_data ||= begin`)?